src/main/java/http/server/handlers/MethodNotAllowedHandler.java

```diff
@@ -0,0 +1,38 @@
+package http.server.handlers;
+
+import http.server.Handler;
+import http.server.Response;
+
+import java.util.*;
+
+public class MethodNotAllowedHandler implements Handler {
+    private final Map<String, Handler> methodHandlers;
+
+    public MethodNotAllowedHandler(Map<String, Handler> methodHandlers) {
+        this.methodHandlers = methodHandlers;
```

Comment:
Why `MethodNotAllowedHandler` needs `methodHandlers` (plural)? it is a handler by itself, is receiving other handlers at its creation the best decision here? Please correct me if I didn't understand your intention here.